### DISCUSSION TITLE
How to Enable Path Autocomplete in Zed?

### DISCUSSION THREAD

**Discussion by nstlgy** [zed-industries/zed#10252] [Feature Requests]
I'm used to VS Code's path suggestion feature where typing `src = '/'` automatically brings up file/directory options. Does Zed have similar functionality?

This is how it looks in vs code 

https://github.com/zed-industries/zed/assets/122057887/0ad703ec-5435-4eea-8b88-db43ac754e4e

This is how it is for me currently in zed 

https://github.com/zed-industries/zed/assets/122057887/28638b46-857a-41cd-af29-83234baad8f1

Is this possible natively with Zed, or are there extensions that offer this capability?

**Comment by auspy** [comment]
maybe this option is still not available. pls create an issue and maybe it will be solved soon @luvsense

**Reply by mocenigo** [reply]
> maybe this option is still not available. pls create an issue and maybe it will be solved soon @luvsense

Exactly. This is a discussion, not a feature request.

**Comment by tednaaa** [comment]
I searched before, but didn't found an answer, is there something like `trigger autocomplete` similiar vscode?

<img width="1056" alt="image" src="https://github.com/zed-industries/zed/assets/79831859/93dc22b5-7920-4edb-8fe9-b12ed9a9804d">

**Comment by HamzaFarhan** [comment]
Anything on this?

**Comment by shine-on-me** [comment]
>  Screen.Recording.2024-04-07.at.7.58.28.PM.mov 

Hey, how did you get the HTML tag autocomplete to work in Zed? From your recording it automatically prepares the whole `<img src="" alt="" />` tag while for me it does nothing. The Path autocomplete does work for me, typing `/` inside the `src="|"` does show path suggestions.

I'm running Zed v0.180.2 on Ubuntu 24.04.02, fresh install, only the HTML extension is enabled, no other settings changed.

**Reply by NoahCounsell** [reply]
To fix the HTML tag autocomplete problem install the Emmet extension.

<img width="1012" height="127" alt="Screenshot 2025-10-16 at 2 17 33 pm" src="https://github.com/user-attachments/assets/11d73adc-4246-4438-b3ad-9693616190f3" />

**Comment by alshdavid** [comment]
+1 this is something I use a lot in VSCode and miss sorely in Zed

**Comment by 3KINGZ** [comment]
it seems the path intellisense only works on html files for other files like .go it doesn't come up

**Comment by Keyruu** [comment]
Is this really not a feature in Zed? That is a big oversight imo...

**Reply by mocenigo** [reply]
It's open source! Implement it! ;-)

**Reply by Keyruu** [reply]
I'm not that good at Rust unfortunately

**Comment by cboecking** [comment]
how can you expect to create anything meaningful if you cannot even compose a README.md. I do not wish to use AI to simply find a file. I just want to type './' and press ctrl+space.

Please please fix this!

**Reply by mocenigo** [reply]
> how can you expect to create anything meaningful if you cannot even compose a README.md. I do not wish to use AI to simply find a file. I just want to type './' and press ctrl+space.
> 
> Please please fix this!

Learn manners and ask the developers again gently. They owe you nothing, and yet you get a free editor.  Or you can implement the feature yourself and give back.

**Reply by cboecking** [reply]
Confirmed and apologies. I spent way too much time on this topic this morning.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by byte-sized-emi** [comment]
For people finding this on search engines, there's an extension that implements this (unofficial): https://zed.dev/extensions/path-server-lsp
Has been working fine for me, but I've heard that some people are having issues with it.
I think one could make a solid case as to why this should be in the editor (speed, zed already has all the information necessary, ease-of-use for new people, highlighting & code navigation, etc)

**Comment by cboecking** [comment]
my 2 cents... I cannot and will not use zed until this works in the core. This is just my humble opinion and my personal truth. I am struggling to understand how an editor cannot create a simple markdown file with relative links. I want to use it...
